app/code/core/Mage/Core/Model/Design/Package.php

```diff
@@ -672,15 +672,16 @@ protected function _checkUserAgentAgainstRegexps($regexpsConfigPath)
     public static function getPackageByUserAgent(array $rules, $regexpsConfigPath = 'path_mock')
     {
         foreach ($rules as $rule) {
-            if (!empty(self::$_regexMatchCache[$rule['regexp']][$_SERVER['HTTP_USER_AGENT']])) {
+            $userAgent = !empty($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : '';
```

Comment:
It would be shorter (and without warning because ?? performs an isset check) and in my opinion more readable to use: 
```suggestion
            $userAgent = $_SERVER['HTTP_USER_AGENT'] ?? '';
```

Reply:
empty() does an isset check too. See https://www.php.net/manual/en/function.empty.php: "Determine whether a variable is considered to be empty. A variable is considered empty if it does not exist or if its value equals false. **empty() does not generate a warning if the variable does not exist.**"

But yes, it's more readable with ??. Thank you!